api_v3/services/UserService.php

```diff
@@ -583,4 +583,39 @@ public function serveCsvAction($id)
 		$file_path = ExportCsvService::generateCsvPath($id, $this->getKs());
 		return $this->dumpFile($file_path, 'text/csv');
 	}
+
+	/**
+	 * get QR image content
+	 *
+	 * @action generateQrCode
+	 * @param string $hashKey
+	 * @return string
+	 *
+	 */
+	public function generateQrCodeAction($hashKey)
+	{
+		try
+		{
+			$loginData = UserLoginDataPeer::isHashKeyValid($hashKey);
+			if ($loginData)
+			{
+				$this->validateApiAccessControl($loginData->getLastLoginPartnerId());
+			}
+			$dbUser = kuserPeer::getAdminUser($loginData->getConfigPartnerId(), $loginData);
+		}
+		catch (kUserException $e)
+		{
+			throw new KalturaAPIException(KalturaErrors::INVALID_HASH);
+		}
+
+		if (!$dbUser)
+			throw new KalturaAPIException(KalturaErrors::INVALID_USER_ID, $loginData->getLoginEmail());
+		$imgContent = authenticationUtils::getQRImage($dbUser);
+		if(!$imgContent)
+		{
+			throw new KalturaAPIException(KalturaErrors::ERROR_IN_QR_GENERATION);
+		}
+		return $imgContent;
```

Comment:
If the intention is that this call with render a jpg image with the QR code , then need to use string renderrer in the response.